packages/@react-aria/table/docs/useTable.mdx

```diff
@@ -800,6 +806,696 @@ let rows = [
+        type="tbody">
+        {[...collection.body.childNodes].map(row => (
+          <ResizableTableRow key={row.key} item={row} state={state}>
+            {[...row.childNodes].map(cell => (
+              <ResizableTableCell
+                key={cell.key}
+                cell={cell}
+                state={state}
+                /*- begin highlight -*/
+                widths={widths}
+                /*- end highlight -*/
+              />
+            ))}
+          </ResizableTableRow>
+        ))}
+      </ResizableTableRowGroup>
+    </table>
+  );
+}
+```
+
+</details>
+
+### Resizable table header
+
+The `TableRowGroup` and `TableHeaderRow` only require minor style changes to accommodate the new `display` changes made in
+`ResizableColumnsTable`.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+function ResizableTableRowGroup({type: Element, style, children}) {
+  let {rowGroupProps} = useTableRowGroup();
+  return (
+    <Element
+      {...rowGroupProps}
+      style={{
+        /*- begin highlight -*/
+        display: 'block',
+        /*- end highlight -*/
+        ...style
+      }}>
+      {children}
+    </Element>
+  );
+}
+```
+</details>
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+function ResizableTableHeaderRow({item, state, children}) {
+  let ref = useRef();
+  let {rowProps} = useTableHeaderRow({node: item}, state, ref);
+
+  return (
+    <tr
+      {...rowProps}
+      ref={ref}
+      /*- begin highlight -*/
+      style={{display: 'flex'}}
+      /*- end highlight -*/
+    >
+      {children}
+    </tr>
+  );
+}
+```
+
+</details>
+
+The `TableColumnHeader` is where we see the bulk of the changes required to support resizable columns. First of all, we need to accommodate a `Resizer` element in every resizable column that
+the user can drag or focus to perform a resize operation. To avoid interfering with the existing grid keyboard navigation, we only display this resizer when
+the column is hovered or if the column itself is being resized. This however introduces another issue: how will keyboard, screen reader, and touch users begin a resize operation if the resizer
+is only visible on hover? We could have the user press on the column header to trigger a resize operation, but that action is often reserved for sorting and may feel unexpected to a user.
+
+To resolve this, we need to render a menu button in the column header if the column is resizable, allowing non-mouse users to select from a list of available actions on the column. For resizing, selecting the corresponding option
+should cause focus to shift to the resizer itself, displaying the resizer for touch screen users to drag and letting keyboard and screen readers resize the column via keyboard arrows and screen reader operations
+accordingly.
+
```

Comment:
The issue with this was grid navigation. When arrow key navigating, you couldn't move between column headers. Instead, it'd start resizing.

Reply:
https://github.com/adobe/react-spectrum/pull/3982#discussion_r1091336950 alludes to this issue. When using the arrow keys to navigate between columns, you'll actually move focus to the resizer instead which means you are now in resize mode.

Reply:
hmm maybe we should have a mode where only the up/down arrows change the size rather than left/right? the menu solution is cool, but kinda spectrum design specific. I could see this being a frequent request.

Reply:
that sounds reasonable, hadn't thought to divide up the actions like that

Reply:
I'll see if I can put a PR up for this as

Reply:
Ran into a snag trying to implement this. Since we `setKeyboardNavigationDisabled(true)` when the resizer is focused, I need to re-enable it when the user presses the left/right arrow keys. However, `useGridCell` `onKeyDownCapture` triggers before the `onKeyDownCapture` I setup in `useTableColumnResizing`, meaning we don't move focus back to the column header when on `ArrowLeft`.

Reply:
Similar issue happens if I get rid of `setKeyboardNavigationDisabled` entirely instead. Left/right arrow keys can be made to move between the resizer and the column but up/down arrows keys are then handled by `useGridCell` and results in focus moving from the resizer to the cell below.

Reply:
ugh, yeah, that wouldn't be separable. keyboard grid navigation is either all or nothing... there's no way to distinguish between the two. we'd need to introduce a new prop or something and even then I'm not sure we could get ahead of it, as you mentioned you ran into a problem getting ahead of the capture events

Reply:
could we change it from onFocus to on 'Return/Enter' pressed? maybe that could be the new prop?

Reply:
@devongovett any opinions on the above? Right now these are the options that I can see:
- make it so that "editMode" is enabled when the user presses Enter when focused on the resizer
  - perhaps non-intuitive?
- change the aria example so pressing enter on the column header enters resize mode
  - this doesn't solve grid navigation bit but it gets rid of the menu

Reply:
Either would be ok I think. Just need to make sure it's obvious somehow. Definitely would need an aria-label on the column header if that's how we go, but even for non-screen reader users it might not be obvious. Enter on the resizer itself is a little more I guess.

Reply:
pushed code changes for pressing enter on the column header to enter resize mode (note that I haven't updated the docs copy to reflect the new example code). I'll open a separate branch for the "editMode" updates since that will require changes to the hooks themselves